### PR TITLE
Unified External Aero Recipe: mesh augmentations follow their config; rename/drop reach domain-level global_data

### PR DESCRIPTION
**Pull request by melo-gonzo**
**Summary**

Three config-driven transforms in `physicsnemo.datapipes.transforms.mesh` did not do what their
YAML in the unified external aero recipe says. Found while adding a new dataset to the recipe.

1. `RandomRotateMesh(axes=[...])` without an explicit `mode` rotated over all of SO(3): the
   default `mode="uniform"` ignored `axes`. `mode` now defaults to `"axis_aligned"` when `axes`
   is given and `"uniform"` otherwise. `axes` with `mode="uniform"` raises `ValueError`. The default
   behavior is still to do uniform rotations. 
2. The recipe's `RandomTranslateMesh` block passed plain lists as `low` / `high` to
   `torch.distributions.Uniform`, which accepts only numbers or tensors, so the block failed to
   instantiate whenever `augment: true`. The transform is unchanged; the YAMLs now build the
   bounds with `_target_: torch.tensor`, which keeps `distribution` as the only constructor argument
   and works for any `torch.distributions` class.
3. `RenameMeshFields` and `DropMeshFields` applied their `global_data` mapping to each
   sub-mesh but not to the `DomainMesh` domain-level `global_data`, which is where
   `NonDimensionalizeByMetadata` reads the freestream constants. Both now override
   `apply_to_domain` to cover the domain level. Complements #1961, which routed nested leaf names.

The six unified-aero dataset YAMLs and the README snippet set `mode: axis_aligned` and tensor
translation bounds, so the configured augmentation is the one that runs.

**Changes**
- `physicsnemo/datapipes/transforms/mesh/augmentations.py`: `mode=None` default resolved from
  `axes`; error on `axes` + `uniform`.
- `physicsnemo/datapipes/transforms/mesh/transforms.py`: `apply_to_domain` on
  `RenameMeshFields` and `DropMeshFields`.
- `test/datapipes/transforms/test_mesh_augmentations.py`: axes without mode is axis-aligned;
  no axes defaults to uniform; axes + uniform raises.
- `test/datapipes/transforms/test_nested_fields.py`: domain-level rename and drop.
- Six `examples/cfd/external_aerodynamics/unified_external_aero_recipe/datasets/*.yaml`, README.
- `CHANGELOG.md`.

**Compatibility**
`RandomRotateMesh(axes=...)` with no `mode` now produces axis-aligned rotations. Code that passed
`axes` and depended on the undocumented SO(3) fallback changes; code that does not pass `axes`
is unaffected. `RandomTranslateMesh` is unchanged.

**Testing**
`test/datapipes`: 745 passed, 126 skipped (CPU-only run) on the branch head. Each recipe dataset YAML's `augmentations`
block instantiates through `hydra.utils.instantiate` and samples in-plane offsets with zero z.

### CODE REVIEW

File: physicsnemo/datapipes/transforms/mesh/augmentations.py [hunk 1]
```diff
@@ -208,8 +231,23 @@ def __init__(
...
+        if (low is None) != (high is None):
+            raise ValueError("low and high must be given together")
+        if low is not None:
+            if distribution is not None:
+                raise ValueError("low/high cannot be combined with distribution")
+            ### Uniform rejects lists (and OmegaConf's ListConfig), which is
+            ### what config-driven instantiation hands us.
+            distribution = torch.distributions.Uniform(
+                _as_bound_tensor(low), _as_bound_tensor(high)
+            )
```
[the remark this answers is not in the document]
Comment (melo-gonzo) [reply physicsnemo/datapipes/transforms/mesh/augmentations.py]: Correct issue - fixed in 0a372531.

File: physicsnemo/datapipes/transforms/mesh/augmentations.py [hunk 2]
```diff
@@ -317,6 +361,12 @@ def __init__(
             If ``True``, transform global-data fields under rotation.
         """
         super().__init__()
+        if mode is None:
+            ### Passing axes without a mode is a request for axis-aligned
+            ### rotations; silently falling back to SO(3) would ignore them.
+            mode = "axis_aligned" if axes is not None else "uniform"
```
[the remark this answers is not in the document]
Comment (melo-gonzo) [reply physicsnemo/datapipes/transforms/mesh/augmentations.py:330]: Correct issue - fixed in 0a372531.

**Comment by peterdsharpe** [comment]
I think that `RandomRotateMesh` should default to uniform (current behavior), rather than axis-aligned (this PR), for a few reasons:
- Axis-aligned imposes a bias towards an (arbitrary) coordinate system, which is ultimately nonphysical - there is nothing physically special about $\hat{x}, \hat{y}, \hat{z}$ as a basis set for rotation vectors
- Uniform is more diverse (it's a strictly-larger set than axis-aligned)
- "Random rotation" to me implies equal measure ([Harr-uniform](https://en.wikipedia.org/wiki/Haar_measure)) across the "unit sphere" of rotations, so I think defaulting to axis-aligned would be "surprising" to users

Happy to discuss more offline!


EDIT: Resolved offline - let's keep this PR's version.

**Comment by peterdsharpe** [comment]
Likewise, on translations - it would be nice to not bake in that these random translations are uniform by default, as part of the datapipe. For example, a Cauchy or a Gaussian distribution would be nice if one wants to avoid a hard cutoff. Therefore, it'd be nice if we can avoid any Uniform-specific kwargs in the transform signature itself.

It seems like the true issue here is that the existing method with `_target_` was not working - though it seems to work in my invocations? Let's chat and see if we can figure out where the disconnect is.

**Comment by peterdsharpe** [comment]
On item 3:

> RenameMeshFields and DropMeshFields applied their global_data mapping to each
sub-mesh but not to the DomainMesh domain-level global_data, which is where
NonDimensionalizeByMetadata reads the freestream constants. Both now override
apply_to_domain to cover the domain level.

This part of the edits look good to me! I think this is similar to what we did in https://github.com/NVIDIA/physicsnemo/pull/1961; just linking that here for cross-discoverability.

**Review by peterdsharpe — CHANGES_REQUESTED** [review]
See earlier comments - happy to chat about these offline

**Review by peterdsharpe — APPROVED** [review]
Thanks for changes, LGTM now! One last thing, non-blocking - if you don't mind either editing the top-level PR body text, or dropping a short comment, reflecting the tweaks we made (e.g., the `torch.tensor` target hack), that'd be good for the record. Nice work!

**Comment by melo-gonzo** [comment]
Thank you @peterdsharpe!

Changes made during review:
- `RandomTranslateMesh` reverted to upstream; `distribution` remains the only argument.
- Recipe YAMLs and README wrap the `Uniform` bounds in `_target_: torch.tensor` nodes.
- Changelog entry updated to match additions
- Rotation default unchanged: `RandomRotateMesh()` with no arguments is still uniform.

**Review by coreyjadams — APPROVED** [review]
LGTM
